pkg/util/value_by_path.go

```diff
@@ -0,0 +1,41 @@
+package util
+
+import (
+	"fmt"
+	"strings"
+)
+
+// GetValueByPath retrieves a value from a nested map using a dot-separated path
+func GetValueByPath(data map[string]interface{}, path string) (interface{}, error) {
```

Comment:
suggestion: would you consider using `unstructured` helpers from kubernetes/apimachinery? There is a [`NestedFieldCopy`](https://github.com/kubernetes/apimachinery/blob/d794766488ac2892197a7cc8d0b4b46b0edbda80/pkg/apis/meta/v1/unstructured/helpers.go#L39-L45) function which is used for walking through yaml data, similar to the intent here. Given KEDA already depends on kubernetes/apimachinery, it might be good to reuse the exported code if it suits well.

Reply:
thanks for checking, I agree with your findings